### PR TITLE
EN 8234 add unit tests

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added storage epoch start bootstrap unit tests

### CODE REVIEW

**Review by raduchis — APPROVED** [review]

File: dataRetriever/storageResolvers/trieNodeResolver_test.go [hunk 1]
```diff
@@ -1,3 +1,158 @@
...
+	t.Parallel()
+
+	args := createMockTrieResolverArguments()
+	buff := []byte("data")
+	args.TrieDataGetter = &mock.TrieStub{
+		GetSerializedNodesCalled: func(bytes []byte, u uint64) ([][]byte, uint64, error) {
+			return [][]byte{buff}, 1, nil
+		},
+	}
+	numSendCalled := uint32(0)
```
Comment (bogdan-rosianu) [inline dataRetriever/storageResolvers/trieNodeResolver_test.go:111]: rename to `countSendToConnectedPeerCalled`

File: dataRetriever/storageResolvers/trieNodeResolver_test.go [hunk 2]
```diff
@@ -1,3 +1,158 @@
...
+	args := createMockTrieResolverArguments()
+	buff := []byte("data")
+	numGetSerializedCalled := uint32(0)
+	args.TrieDataGetter = &mock.TrieStub{
+		GetSerializedNodesCalled: func(bytes []byte, u uint64) ([][]byte, uint64, error) {
+			atomic.AddUint32(&numGetSerializedCalled, 1)
+			return [][]byte{buff}, 1, nil
+		},
+	}
+	numSendCalled := uint32(0)
```
Comment (bogdan-rosianu) [inline dataRetriever/storageResolvers/trieNodeResolver_test.go:139]: rename to `countSendToConnectedPeerCalled`

File: dataRetriever/storageResolvers/trieNodeResolver_test.go [hunk 3]
```diff
@@ -1,3 +1,158 @@
...
+	assert.Equal(t, 0, len(args.ChanGracefullyClose))
+	assert.Equal(t, uint32(1), atomic.LoadUint32(&numSendCalled))
+}
+
+func TestTrieNodeResolver_RequestDataFromHashArrayShouldWork(t *testing.T) {
+	t.Parallel()
+
+	args := createMockTrieResolverArguments()
+	buff := []byte("data")
+	numGetSerializedCalled := uint32(0)
```
Comment (bogdan-rosianu) [inline dataRetriever/storageResolvers/trieNodeResolver_test.go:132]: rename to `countGetSerializedNodesCalled`

File: epochStart/bootstrap/epochStartMetaBlockProcessor.go [hunk 4]
```diff
@@ -128,7 +128,12 @@ func (e *epochStartMetaBlockProcessor) Save(data process.InterceptedData, fromCo
 		return nil
 	}
 
-	metaBlock := interceptedHdr.HeaderHandler().(*block.MetaBlock)
+	metaBlock, ok := interceptedHdr.HeaderHandler().(*block.MetaBlock)
+	if !ok {
+		log.Warn("saving epoch start meta block error", "error", epochStart.ErrWrongTypeAssertion)
```
Comment (bogdan-rosianu) [inline epochStart/bootstrap/epochStartMetaBlockProcessor.go:133]: you could also display the `interceptedHdr.HeaderHandler()` for debugging purposes.

File: epochStart/bootstrap/process.go [hunk 5]
```diff
@@ -39,7 +39,8 @@ import (
 
 var log = logger.GetOrCreate("epochStart/bootstrap")
 
-const timeToWait = time.Minute
+// DefaultTimeToWait represents the default timespan until requested data needs to be received from the connected peers
+const DefaultTimeToWait = time.Minute
```
Comment (bogdan-rosianu) [inline epochStart/bootstrap/process.go:43]: maybe rename to `DefaultTimeToWaitForRequestedData`

File: epochStart/bootstrap/storageEpochStartMetaBlockProcessor.go [hunk 6]
```diff
@@ -80,7 +80,12 @@ func (ses *storageEpochStartMetaBlockProcessor) Save(data process.InterceptedDat
 		return nil
 	}
 
-	metaBlock := interceptedHdr.HeaderHandler().(*block.MetaBlock)
+	metaBlock, ok := interceptedHdr.HeaderHandler().(*block.MetaBlock)
+	if !ok {
+		log.Warn("saving epoch start meta block error", "error", epochStart.ErrWrongTypeAssertion)
```
Comment (bogdan-rosianu) [inline epochStart/bootstrap/storageEpochStartMetaBlockProcessor.go:85]: also log the header handler

File: epochStart/bootstrap/storageProcess.go [hunk 7]
```diff
@@ -38,6 +40,7 @@ type storageEpochStartBootstrap struct {
 	importDbConfig      config.ImportDbConfig
 	chanGracefullyClose chan endProcess.ArgEndProcess
 	chainID             string
+	timeToWait          time.Duration
```
Comment (bogdan-rosianu) [inline epochStart/bootstrap/storageProcess.go:43]: `timeToWaitForRequestedData` ?

File: factory/bootstrapComponents.go [hunk 8]
```diff
@@ -164,6 +164,7 @@ func (bcf *bootstrapComponentsFactory) Create() (*bootstrapComponents, error) {
 			ArgsEpochStartBootstrap: epochStartBootstrapArgs,
 			ImportDbConfig:          bcf.importDbConfig,
 			ChanGracefullyClose:     bcf.coreComponents.ChanStopNodeProcess(),
+			TimeToWait:              bootstrap.DefaultTimeToWait,
```
Comment (bogdan-rosianu) [inline factory/bootstrapComponents.go:167]: `TimeToWaitForRequestedData` ?

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: dataRetriever/storageResolvers/trieNodeResolver_test.go (hunk 1 above)
Comment (iulianpascalau) [reply dataRetriever/storageResolvers/trieNodeResolver_test.go:111]: `numSendToConnectedPeerCalled` as to be consistent across other test files

File: dataRetriever/storageResolvers/trieNodeResolver_test.go (hunk 2 above)
Comment (iulianpascalau) [reply dataRetriever/storageResolvers/trieNodeResolver_test.go:139]: `numSendToConnectedPeerCalled`

File: dataRetriever/storageResolvers/trieNodeResolver_test.go (hunk 3 above)
Comment (iulianpascalau) [reply dataRetriever/storageResolvers/trieNodeResolver_test.go:132]: `numGetSerializedNodesCalled`

File: epochStart/bootstrap/epochStartMetaBlockProcessor.go (hunk 4 above)
Comment (iulianpascalau) [reply epochStart/bootstrap/epochStartMetaBlockProcessor.go:133]: done

File: epochStart/bootstrap/process.go (hunk 5 above)
Comment (iulianpascalau) [reply epochStart/bootstrap/process.go:43]: done

File: epochStart/bootstrap/storageEpochStartMetaBlockProcessor.go (hunk 6 above)
Comment (iulianpascalau) [reply epochStart/bootstrap/storageEpochStartMetaBlockProcessor.go:85]: done

File: epochStart/bootstrap/storageProcess.go (hunk 7 above)
Comment (iulianpascalau) [reply epochStart/bootstrap/storageProcess.go:43]: done

File: factory/bootstrapComponents.go (hunk 8 above)
Comment (iulianpascalau) [reply factory/bootstrapComponents.go:167]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**